pkg/chunk/openstack/swift_object_client.go

```diff
@@ -0,0 +1,178 @@
+	f.StringVar(&cfg.UserDomainName, prefix+"swift.user-domain-name", "", "Openstack user's domain name.")
+	f.StringVar(&cfg.UserDomainID, prefix+"swift.user-domain-id", "", "Openstack user's domain id.")
+	f.StringVar(&cfg.Username, prefix+"swift.username", "", "Openstack username for the api.")
+	f.StringVar(&cfg.UserId, prefix+"swift.user-id", "", "Openstack userid for the api.")
+	f.StringVar(&cfg.Password, prefix+"swift.password", "", "Openstack api key.")
+	f.StringVar(&cfg.AuthUrl, prefix+"swift.auth-url", "", "Openstack authentication URL.")
+	f.StringVar(&cfg.RegionName, prefix+"swift.region-name", "", "Openstack Region to use eg LON, ORD - default is use first region (v2,v3 auth only)")
+	f.StringVar(&cfg.ProjectName, prefix+"swift.project-name", "", "Openstack project name (v2,v3 auth only).")
+	f.StringVar(&cfg.ProjectID, prefix+"swift.project-id", "", "Openstack project id (v2,v3 auth only).")
+	f.StringVar(&cfg.ProjectDomainName, prefix+"swift.project-domain-name", "", "Name of the project's domain (v3 auth only), only needed if it differs from the user domain.")
+	f.StringVar(&cfg.ProjectDomainID, prefix+"swift.project-domain-id", "", "Id of the project's domain (v3 auth only), only needed if it differs the from user domain.")
+}
+
+// NewSwiftObjectClient makes a new chunk.Client that writes chunks to OpenStack Swift.
+func NewSwiftObjectClient(cfg SwiftConfig, delimiter string) (*SwiftObjectClient, error) {
+	util.WarnExperimentalUse("OpenStack Swift Storage")
+
+	// Create a connection
+	c := &swift.Connection{
+		AuthUrl:  cfg.AuthUrl,
+		ApiKey:   cfg.Password,
+		UserName: cfg.Username,
+		UserId:   cfg.UserId,
+
+		TenantId:       cfg.ProjectID,
+		Tenant:         cfg.ProjectName,
+		TenantDomain:   cfg.ProjectDomainName,
+		TenantDomainId: cfg.ProjectDomainID,
+
+		Domain:   cfg.DomainName,
+		DomainId: cfg.DomainId,
+
+		Region: cfg.RegionName,
+	}
+
+	switch {
+	case cfg.UserDomainName != "":
+		c.Domain = cfg.UserDomainName
+	case cfg.UserDomainID != "":
+		c.DomainId = cfg.UserDomainID
+	}
+
+	if len(delimiter) > 1 {
+		return nil, fmt.Errorf("delimiter must be a single character but was %s", delimiter)
+	}
+	var delim rune
+	if len(delimiter) != 0 {
+		delim = []rune(delimiter)[0]
+	}
+
+	// Authenticate
+	err := c.Authenticate()
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure the container is created, no error is returned if it already exists.
+	if err := c.ContainerCreate(cfg.ContainerName, nil); err != nil {
+		return nil, err
+	}
+
+	return &SwiftObjectClient{
+		conn:      c,
+		cfg:       cfg,
+		delimiter: delim,
+	}, nil
+}
+
+func (s *SwiftObjectClient) Stop() {
+	s.conn.UnAuthenticate()
+}
+
+// GetObject returns a reader for the specified object key from the configured swift container. If the
+// key does not exist a generic chunk.ErrStorageObjectNotFound error is returned.
+func (s *SwiftObjectClient) GetObject(ctx context.Context, objectKey string) (io.ReadCloser, error) {
```

Comment:
The fact that `ctx` parameter is not used in any of `GetObject`, `PutObject`, `List` or `DeleteObject` methods is concerning.